server/dining/balance.py

```diff
@@ -120,6 +122,8 @@ def get_average_balances_by_day():
             })
 
         df = pd.DataFrame(balance_array).groupby('timestamp').agg(lambda x: x.mean()).reset_index()
+        if startDate and endDate:
+            df = df[(df['timestamp'] > startDate) & (df['timestamp'] < endDate)]
```

Comment:
We should be grouping by date instead of timestamp, correct? If we group by timestamp, you could still get multiple values for the same day.

Reply:
the timestamp is in %Y-%M-%D without the hours and minutes so there wouldn't be multiple timestamp when we group by day

Reply:
I see, cool!